crates/symbolicator/src/services/download/s3.rs

```diff
@@ -116,37 +120,45 @@ impl S3Downloader {
             );
             let s3 = Arc::new(match key.aws_credentials_provider {
                 AwsCredentialsProvider::Container => {
-                    let container_provider = rusoto_credential::ContainerProvider::new();
-                    let provider =
-                        match rusoto_credential::AutoRefreshingProvider::new(container_provider) {
-                            Ok(provider) => provider,
-                            Err(err) => panic!(
-                            "Unable to instantiate rusoto_credential::AutoRefreshingProvider: {:?}",
-                            err
-                        ),
-                        };
-                    self.create_s3_client(provider, region)
+                    let provider = LazyCachingCredentialsProvider::builder()
+                        .load(provide_credentials_fn(|| async {
+                            aws_config::ecs::EcsCredentialsProvider::builder()
+                                .build()
+                                .credentials()
+                                .await
+                        }))
+                        .build();
+                    self.create_s3_client(provider, region).await
                 }
                 AwsCredentialsProvider::Static => {
-                    let provider = rusoto_credential::StaticProvider::new_minimal(
+                    let provider = Credentials::from_keys(
                         key.access_key.clone(),
                         key.secret_key.clone(),
+                        None,
                     );
-                    self.create_s3_client(provider, region)
+                    self.create_s3_client(provider, region).await
                 }
             });
 
-            container.put(key.clone(), s3.clone());
+            {
+                let mut container = self.client_cache.lock();
+                container.put(key.clone(), s3.clone())
+            };
             s3
         }
     }
 
-    fn create_s3_client<P: ProvideAwsCredentials + Send + Sync + 'static>(
+    async fn create_s3_client<P: ProvideCredentials + Send + Sync + 'static>(
         &self,
         provider: P,
```

Comment:
Why not use this simpler syntax instead?
```rust
    async fn create_s3_client(
        &self,
        provider: impl ProvideCredentials + Send + Sync + 'static,
        region: Region,
    ) -> Client {
        let shared_config = aws_config::from_env()
            .credentials_provider(provider)
            .region(region)
            .load()
            .await;
        Client::new(&shared_config)
    }
```

Reply:
Sure, it might well be that this code was from before `impl trait` was allowed in parameters.  Feel free to change if if you prefer, I don't think there's a particular reason here to prefer the parametrised version.

Reply:
Thanks, commited in [2bf55d0](https://github.com/getsentry/symbolicator/pull/849/commits/2bf55d070bd2764ae47fa50f263e70ab830b56a3).